src/test/graphdb.test.js

```diff
@@ -77,6 +77,198 @@ const SCHEMA_WITH_PROPERTY_AND_EDGE_SAME_NAME = {
+                    "type": "String"
+                },
+                {
+                    "name": "ip_range.last_ip",
+                    "type": "String"
+                }
+            ]
+        },
+        {
+            "label": "abc=123@[efg]456",
+            "properties": [
+                {
+                    "name": "instance_type",
+                    "type": "String"
+                },
+                {
+                    "name": "state",
+                    "type": "String"
+                },
+                {
+                    "name": "arn",
+                    "type": "String"
+                }
+            ]
+        },
+        {
+            "label": "abc{123}|efg-456",
+            "properties": [
+                {
+                    "name": "name",
+                    "type": "String"
+                },
+                {
+                    "name": "ip_range.first_ip",
+                    "type": "String"
+                },
+                {
+                    "name": "ip_range.last_ip",
+                    "type": "String"
+                }
+            ]
+        },
+    ],
+    "edgeStructures": [
+        {
+            "label": "resource_link",
+            "properties": [],
+            "directions": [
+                {
+                    "from": "abc!$123&efg",
+                    "to": "abc(123).efg:456",
+                    "relationship": "MANY-ONE"
+                },
+                {
+                    "from": "abc!$123&efg",
+                    "to": "abc!$123&efg",
+                    "relationship": "ONE-ONE"
+                }
+            ]
+        }
+    ]
+};
+
+const SCHEMA_WITH_SPECIAL_CHARS_RESULT = `
+    type Abc_ex__dol_123_amp_efg @alias(property:\"abc!$123&efg\") {
+        _id: ID! @id
+        instance_type: String
+        state: String
+        arn: String
+        abc_op_123_cp__dot_efg_cn_456Resource_linkOut: Abc_op_123_cp__dot_efg_cn_456 @relationship(edgeType:\"resource_link\", direction:OUT)
+        abc_ex__dol_123_amp_efgResource_linkOut: Abc_ex__dol_123_amp_efg @relationship(edgeType:\"resource_link\", direction:OUT)
+        abc_ex__dol_123_amp_efgResource_linkIn: Abc_ex__dol_123_amp_efg @relationship(edgeType:\"resource_link\", direction:IN)
+        resource_link:Resource_link
+    }
+
+    input Abc_ex__dol_123_amp_efgInput {
+        _id: ID @id
+        instance_type: String
+        state: String
+        arn: String
+    }
+
+    type Abc_op_123_cp__dot_efg_cn_456 @alias(property:\"abc(123).efg:456\") {
+        _id: ID! @id
+        name: String
+        ip_range_dot_first_ip: String @alias(property: \"ip_range.first_ip\")
+        ip_range_dot_last_ip: String @alias(property: \"ip_range.last_ip\")
+        abc_ex__dol_123_amp_efgResource_linksIn(filter: Abc_ex__dol_123_amp_efg Input, options: Options): [Abc_ex__dol_123_amp_efg] @relationship(edgeType:\"resource_link\", direction:IN)
```

Comment:
This test is failing - probably because this expected output contains spaces before `Input`.